PL_2_komentarze.py

```diff
@@ -0,0 +1,2722 @@
+        oknoklient.destroy()
+        ladujklientow()
+
+    def bind(self, k, f):
+        return self.tree.bind(k, f)
+
+def znajdzklienta():
+
+    wprowadzony = entryKlient1.get()
+    listaponumerowana = list(enumerate(listaklientow))
+    znalezioneindeksy = [i for i,x in listaponumerowana if wprowadzony.lower() in x.lower()]
+    rezultatyklient = []
+    if not znalezioneindeksy:
+        messagebox.showinfo("Nie znaleziono klienta!", "Nie udało się dopasować wprowadzonej frazy do żadnego z klientów.\nWypełnij pola adresowe ręcznie, a następnie kliknij przycisk 'Dodaj klienta', aby zachować dane na przyszłość.")
+    else:
+        global oknoklient
+        oknoklient = Toplevel(root)
+        oknoklient.iconbitmap(r'bin\tubes.ico')
+        oknoklient.title('Klienci zawierający szukane słowo:')
+        for item in znalezioneindeksy:
+            wartoscadresu1 = wsX.cell(item,1).value
+            key = b'Uf-FryHfa-JXDToRDClFsSrv2h56kzEUCK5imvQngbo='
+            f = Fernet(key)
+            kodowanie1 = wartoscadresu1.encode()
+            odkryte1 = f.decrypt(kodowanie1)
+            wartoscadresu11 = odkryte1.decode('utf-8')
+            wartoscadresu2 = wsX.cell(item,2).value
+            kodowanie2 = wartoscadresu2.encode()
+            odkryte2 = f.decrypt(kodowanie2)
+            wartoscadresu22 = odkryte2.decode('utf-8')
+            rezultatyklient.append((listaklientow[item], wartoscadresu11, wartoscadresu22))
+        
+        Drzewo(oknoklient, rezultatyklient, entryKlient1, entryKlient2, entryKlient3)
+        Tooltip(oknoklient, text='Potwierdź wybór klienta podwójnym kliknięciem')
+    
+        if len(rezultatyklient) == 1:
+            tupelek = rezultatyklient[0]
+            entryKlient1.delete(0, END)
+            entryKlient1.insert(END, tupelek[0])
+            entryKlient1.validate()
+            entryKlient2.delete(0, END)
+            entryKlient2.insert(END, tupelek[1])
+            entryKlient2.validate()
+            entryKlient3.delete(0, END)
+            entryKlient3.insert(END, tupelek[2])
+            entryKlient3.validate()
+            oknoklient.destroy()
+            ladujklientow()
+
+def znajdzklientaenter(event):
+    znajdzklienta()
+
+ludzie = {
+    'adarog': 0,
+    'damsre': 1,
+    "arkkla": 2,
+    "marwec": 3,
+    "micros": 4,
+    "robtom": 5,
+    "krzwen": 6,
+    "hubsta": 7,
+    "danjoz": 8,
+    "krzszr": 0,
+    "malsmu": 1,
+    "marfal": 2,
+    "robwoz": 3,
+    "agnbod": 0,
+    "jarwal": 1,
+    "tomrec": 2,
+    "sylpik": 0,
+    "wojlej": 1,
+    "arkgaj": 2,
+    "miczah": 3,
+    "piosik": 4,
+    "piomur": 0,
+    "pawsko": 1,
+    "martry": 2,
+    "joasta": 3,
+    "tomnap": 0,
+    "eugpod": 1,
+    "darken": 0,
+    "przkap": 1,
+    "tomgrz": 0,
+    "jaknec": 1,
+    "pawpud": 2
+    }
+
+
+
+def kimjestes():
+    try:
+        entryprzygotowal.current(ludzie.get(getlogin()))
+    except:
+        pass
+
+def ladujlistewezy(s):
+    if s == 'PL':
```

Comment:
1. Powtórzenie kodu, jedyny element różniący wnętrze obu ifów to ścieżka do arkusza. Śmiało można wydzielić jej generowanie do osobnej, małej funkcji. 
2. Generalnie lepiej mieć więcej małych funkcji, niż mniej dużych. Jest taka zasada Single Responsibility Principle mówiąca, że każda funkcja/metoda powinna mieć tylko jeden cel swojego działania. U Ciebie `ladujlistewezy(s)` robi dwie rzeczy: 
sprawdza język
ładuje listę
Po podziale na dwie osobne funkcje kod zyskuje na czytelności, spada ryzyko że walnie error w miejscu niespodziewanym. Teraz widzę co powinien zwrócić `print` w `else`, coś w stylu `Unknown language, cannot load hose list`